src/abaci.sol

```diff
@@ -15,7 +17,7 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-pragma solidity >=0.6.12;
```

Comment:
I may have missed a discussion around this, but locking these may break any repos that accept this as a dependency that are using a later (but compatible) version. I don't think this should go do master.

Reply:
I thought about the same thing, and was about to mention it. But solidity `0.6.12` is the last one and haven't been a new one for a long while. So at this stage seems very unlikely a new one happens, then is the same in practice.

Reply:
@naszam made me aware of this best practice advice by ConsenSys: https://consensys.github.io/smart-contract-best-practices/development-recommendations/solidity-specific/locking-pragmas/

Initially I was favor of something like `~0.6.12` which allows for a patch bump, but the problem is that it also compiles with nightly builds.

Reply:
yeah, this was discussed in https://github.com/makerdao/MIP21-RWA-Example/pull/7 as well.

Reply:
Well let's not compare MIP21 with this repo that they have nothing to do between.
This repo is imported in several other ones. Locking to a version might provoke a nightmare, if it wasn't that we are in the latest version and we do not think there will be a new one I would be in opposition to it.
Best practices are subjective, they are also relative to the situation. So sometimes what Slither or Consensys says doesn't really matter.
Also IMO, this one kind of lacks of real value, the pragma for example doesn't ensure you are compiling with the specific intended flags.
If you want to make sure to be compiling exactly as was tested then build a script, use a docker image or nix procedure to make sure it is being done as corresponds.

Reply:
The discussion I was referring to is https://github.com/makerdao/MIP21-RWA-Example/pull/7#discussion_r809173471, where a locked pragma `0.5.12` was replaced by `~0.6.12`.
Regarding pragma best practises, as stated in the [SWC-13](https://swcregistry.io/docs/SWC-103):
> Pragma statements can be allowed to float when a contract is intended for consumption by other developers, as in the case with contracts in a library or EthPM package. Otherwise, the developer would need to manually update the pragma in order to compile locally.

So, in case of contract used as dependency in other repos, I think we should keep the floating pragma (e.g. `>=0.6.12`)

Reply:
Fair enough. How about we use the [caret range](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004) `^0.6.12` (which I just realized is the same as a tilde range) so that only a patch bump is allowed

Reply:
There may not be active development on 0.6.x right now but the discovery of a security vulnerability sometimes causes devs to go back and patch earlier versions.

Reply:
done